src/email/services/mailing.service.ts

```diff
@@ -19,4 +26,31 @@ export class MailingService {
       throw new MailSendingException();
     }
   }
+
+  public async sendMailsToSubscribers(): Promise<void> {
+    const { currencyCode, date, rate } =
+      await this.rateService.getExchangeRate();
+
+    const mailMessage = this.mailHtmlTemplate(currencyCode, rate, date);
+    const mailSubject = 'Exchange rate USD to UAH';
+
+    const subscribers = await this.emailService.findAll(true);
+    const mailPromises = subscribers.map((subscriber) =>
+      this.sendMail({
+        html: mailMessage,
+        subject: mailSubject,
+        to: subscriber.email,
+      }),
+    );
+
+    await Promise.all(mailPromises);
```

Comment:
I think, you should consider using Promise.allSettled here, because the sending of each email does not depend on the others. For example, when using Promise.all, if any email throws an error, it can stop the sending of the other emails.